CharacterizeTransformMatrix/CharacterizeTransformMatrix.py

```diff
@@ -0,0 +1,425 @@
+        transformNode = self.ui.inputSelector.currentNode()
+        if transformNode is None:
+            self.ui.transformDescriptionTextEdit.setPlainText(
+                "No transform node selected"
+            )
+            return
+        elif not transformNode.IsLinear():
+            self.ui.transformDescriptionTextEdit.setPlainText(
+                "Selected transform is composite or not linear"
+            )
+            return
+        # Otherwise, transform node exists and is linear
+        results = self.logic.characterizeLinearTransformNode(transformNode)
+        resultsText = results.textResults
+        resultsText = "\n".join(resultsText)
+        self.ui.transformDescriptionTextEdit.setPlainText(resultsText)
+
+    def cleanup(self):
+        """
+        Called when the application closes and the module widget is destroyed.
+        """
+        self.removeObservers()
+
+
+#
+# CharacterizeTransformMatrixLogic
+#
+
+
+class CharacterizeTransformMatrixLogic(ScriptedLoadableModuleLogic):
+    """This class should implement all the actual
+    computation done by your module.  The interface
+    should be such that other python code can import
+    this class and make use of the functionality without
+    requiring an instance of the Widget.
+    Uses ScriptedLoadableModuleLogic base class, available at:
+    https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+    """
+
+    def __init__(self):
+        """
+        Called when the logic class is instantiated. Can be used for initializing member variables.
+        """
+        ScriptedLoadableModuleLogic.__init__(self)
+
+    def characterizeLinearTransformNode(self, transformNode, verbose=True):
+        """A wrapper for characterizeLinearTransformMatrix() which takes a transform node
+        rather than a transform matrix."""
+        H = slicer.util.arrayFromTransformMatrix(transformNode)
+        # H is 4x4 numpy array transformation matrix
+        outputs = self.characterizeLinearTransformMatrix(H, verbose=verbose)
+        return outputs  # outputs is a namedtuple
+
+    def characterizeLinearTransformMatrix(self, H, verbose=True):
+        import numpy as np
+        import scipy
+
+        T, R, K, S, f, X = self.polarDecompose(H)
+        textResults = []
+        #### Stretch
+        # Report scale factors and stretch directions, determine if rigid, largest percent change, volumePercentChange
+        line = "Scale factors and stretch directions (eigenvalues and eigenvectors of stretch matrix K):"
+        textResults.append(line)
+        if verbose:
+            print(line)
+        percentChanges = []
+        scaleDirections = []
+        for idx, (factor, axis) in enumerate(zip(f[:3], X.T[:3])):
+            percentChange = 100 * (factor - 1)
+            percentChanges.append(percentChange)
+            scaleDirections.append(axis)
+            line = "  f%i: %+0.3f%% change in direction [%0.2f, %0.2f, %0.2f]" % (
```

Comment:
Consider using markdown formatting. It makes the generated text more readable even if printed as text, and QTextEdit can render it nicely.